Uncaught DOMException on Marko 5.33.14

### Marko Version: 5.33.14

The issues can be reproduced with this version.

### Details

After the upgrade from 5.33.13, I see an error message in browser console. Website stops rendering.

### Expected Behavior

No errors, as on 5.33.14

### Actual Behavior

Error message:

`Uncaught DOMException: Failed to execute 'createElementNS' on 'Document': The qualified name provided is empty.
    at y.bm_ (https://demo.hereticjs.org/heretic/npm.6e6268443d1e2ee90d8a.js:1:31445)
    at l (https://demo.hereticjs.org/heretic/npm.6e6268443d1e2ee90d8a.js:1:36671)
    at z (https://demo.hereticjs.org/heretic/npm.6e6268443d1e2ee90d8a.js:1:38003)
    at A (https://demo.hereticjs.org/heretic/npm.6e6268443d1e2ee90d8a.js:1:36969)
    at z (https://demo.hereticjs.org/heretic/npm.6e6268443d1e2ee90d8a.js:1:38898)
    at l (https://demo.hereticjs.org/heretic/npm.6e6268443d1e2ee90d8a.js:1:36781)
    at z (https://demo.hereticjs.org/heretic/npm.6e6268443d1e2ee90d8a.js:1:38003)
    at l (https://demo.hereticjs.org/heretic/npm.6e6268443d1e2ee90d8a.js:1:36781)
    at z (https://demo.hereticjs.org/heretic/npm.6e6268443d1e2ee90d8a.js:1:38003)
    at A (https://demo.hereticjs.org/heretic/npm.6e6268443d1e2ee90d8a.js:1:36969)`

### Possible Fix

Downgrade to 5.33.13

<details><summary>Additional Info</summary>

### Your Environment

- Chromium 119.0.6045.199
- Operating System and version (desktop or mobile): MacOS Sonoma 14.4.1
- Link to your project: https://github.com/hereticjsorg/heretic

### Steps to Reproduce

<!-- Provide a link to a live example -->

<!-- or an unambiguous set of steps to reproduce this bug -->

<!-- include code to reproduce, if relevant -->

1.  git clone https://github.com/hereticjsorg/heretic.git
2. npm i && **npm run configure** && npm run build -- --dev _(UPD: forgot to run the configure script)_
3. npm run server
4. Go to http://127.0.0.1:3001 (no error)
5. npm i marko@5.33.14
6. npm run build -- --dev
7. npm run server
8. Error message in browser console will appear

<img width="894" alt="image" src="https://github.com/marko-js/marko/assets/7666940/3d2147ca-cd64-4c1c-9dbc-4a74b4a2c9e9">

### Stack Trace

<!-- If an error is thrown, provide the stack trace here -->

</details>

@xtremespb I'm trying to reproduce but when I run `npm run build -- --dev` I get this
```
> heretic@1.2.10 build-dev
> webpack --mode development --no-color


[webpack-cli] Failed to load '../hereticjsorg/heretic/webpack.config.js' config

[webpack-cli] Error: Cannot find module '../hereticjsorg/heretic/site/etc/languages.json'
```

--- a/.changeset/six-terms-smile.md
+++ b/.changeset/six-terms-smile.md
@@ -1,0 +1,7 @@
+---
+"@marko/compiler": patch
+"marko": patch
+"@marko/translator-default": patch
+---
+
+Fix issues related to hydrating sections under a native tag with the `no-update` directive.

--- a/packages/marko/src/runtime/vdom/morphdom/index.js
+++ b/packages/marko/src/runtime/vdom/morphdom/index.js
@@ -596,9 +596,9 @@ function morphdom(fromNode, toNode, host, componentsContext) {
                 curFromNodeType === TEXT_NODE &&
                 toNextSibling.___nodeType === TEXT_NODE &&
                 curFromNodeValue.startsWith(curToNodeValue) &&
-                toNextSibling.___nodeValue.startsWith(
-                  curFromNodeValue.slice(curToNodeValue.length),
-                )
+                curFromNodeValue
+                  .slice(curToNodeValue.length)
+                  .startsWith(toNextSibling.___nodeValue)
               ) {
                 // In hydrate mode we can use splitText to more efficiently handle
                 // adjacent text vdom nodes that were merged.

